app/src/main/java/com/example/aimovies/domain/use_case/GetFavouriteMovies.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.aimovies.domain.use_case
+
+import com.example.aimovies.data.repository.favourite.FavouriteMovieRepository
+import example.moviedb.FavouriteEntitiy
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Created by A.Elkhami on 25/07/2023.
+ */
+class GetFavouriteMovies(private val repository: FavouriteMovieRepository) {
+    operator fun invoke(): Flow<List<FavouriteEntitiy>> {
```

Comment:
I think this function should be suspend function.

Reply:
This function returns flow and doesn't call any other suspend function, so it doesn't need to be suspended as the flow will do anything related async within itself.